clang/test/CodeGenSYCL/kernelname-enum.cpp

```diff
@@ -145,13 +168,19 @@ int main() {
 // CHECK: enum unscoped_enum : int;
 // CHECK: template <unscoped_enum EnumType> class dummy_functor_6;
 // CHECK: template <typename EnumType> class dummy_functor_7;
-
+// CHECK: namespace type_argument_template_enum {
+// CHECK-NEXT: enum class E : int;
+// CHECK-NEXT: }
+// CHECK: template <type_argument_template_enum::E EnumValue> class T2;
+// CHECK: template <typename T> class T1;
 // CHECK: Specializations of KernelInfo for kernel function types:
-// CHECK: template <> struct KernelInfo<::dummy_functor_1<(no_namespace_int)0>>
-// CHECK: template <> struct KernelInfo<::dummy_functor_2<(no_namespace_short)1>>
-// CHECK: template <> struct KernelInfo<::dummy_functor_3<(internal::namespace_short)1>>
-// CHECK: template <> struct KernelInfo<::dummy_functor_4<(enum_in_anonNS)1>>
-// CHECK: template <> struct KernelInfo<::dummy_functor_5<(no_type_set)0>>
-// CHECK: template <> struct KernelInfo<::dummy_functor_6<(unscoped_enum)0>>
+// CHECK: template <> struct KernelInfo<::dummy_functor_1<(::no_namespace_int)0>>
```

Comment:
I would vastly prefer these be static casts, though feel free to do that in a different patch.

Reply:
Ok. Thanks for the review. I'll open another PR for this.

Reply:
https://github.com/intel/llvm/issues/1784

Created this to keep track of issue. Will work on it once current work is done.